coding-guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,243 @@
+# Coding Guidelines
+
+## Line length
+One Line in JavaScript should have a maximal length of 80 characters.
+
+## Indentation and Line-Breaks
+Rules:
+- Indentation is done with 2 spaces in most cases.
+- Line breaks are only done if needed, try to use the 80 character limit.
+- Strings are splitted at the latest possible position.
+  - At the latest whitespace character.
+  - The second line is indented with 2 spaces.
+- Object Literals are kept on one line if possible.
+  - If the line gets to long put each attribute on a single line.
+  - Multi level object literals should be splitted in to multiple lines. Each level in the object should have 2 spaces as indentation.
+
+Examples:
+```JS
+switch (a) {
+  case "a":
+    break;
+  case "b":
+    break;
+}
+
+var a,
```

Comment:
Hehe, this here is not allowed. We don't combine variable definitions with a comma.
This must be written like this:
```javascript
var a;
var b;
var c;
```
I am sorry. :)